src/Plugin.php

```diff
@@ -185,6 +185,52 @@ public function handlePreInstallUpdateEvent(PackageEvent $event): void
         }
     }
 
+    /**
+     * Handle indirection process, like used by WP EDD
+     * The "indirection" property can contain:
+     * "http" and "ssl" object, as defined by https://github.com/composer/composer/blob/main/src/Composer/Util/Http/CurlDownloader.php
+     * "parse": {
+     *   "format": "json" # only supported value
+     *   "key": # A string (or an object for specifying nested keys to fetch the package download URL from
+     * }
+     */
+    public function fetchIndirection(PreFileDownloadEvent $event, string $url, array $extra): string
+    {
+        $options = [
+            'http' => array_replace_recursive(['method' => 'GET'], $extra['indirection']['http'] ?? []),
+            'ssl' => $extra['indirection']['ssl'] ?? []
+        ];
+
+        $data = $event->getHttpDownloader()->get($url, $options);
+        if ($extra['indirection']['parse']['format'] ?? false === 'json') {
+            $response = $data->decodeJson();
+            $key = $extra['indirection']['parse']['key'] ?? false;
+            if ($key) {
+                // Look for a succession of (nested) keys within a recursive array (from the JSON)
+                $jsonObj = $response;
+                do {
+                    $index = is_string($key) ? $key : key($key);
+                    if (! isset($jsonObj[$index])) {
+                        break;
+                    }
+                    $jsonObj = $jsonObj[$index];
+                    if (!is_array($key)) {
+                        break;
+                    }
+                    $key = $key[$index] ?? false;
+                } while ($key);
+
+                return is_array($jsonObj) ? json_encode($jsonObj) : $jsonObj;
+            }
+
+            // format=json but no key specified (!)
+            return json_encode($response);
+        } else {
+            // raw HTML (future usage of regexp?)
+            return $data->getBody();
+        }
+    }
```

Comment:
```suggestion
    /**
     * Handle indirection process, like used by WP EDD
     *
     * The "indirection" property can contain:
     * "http" and "ssl" object, as defined by https://github.com/composer/composer/blob/main/src/Composer/Util/Http/Curl
     * "parse": {
     *   "format": "json" # only supported value
     *   "key": # A string (or an object for specifying nested keys to fetch the package download URL from
     * }
     */
    public function fetchIndirection(PreFileDownloadEvent $event, string $url, array $extra): string
    {
        $options = [
            'http' => array_replace_recursive(['method' => 'GET'], $extra['indirection']['http'] ?? []),
            'ssl' => $extra['indirection']['ssl'] ?? [],
        ];

        $response = $event->getHttpDownloader()->get($url, $options);

        if ($extra['indirection']['parse']['format'] ?? false !== 'json') {
            // Raw HTML
            // @TODO Future usage of regexp
            return $response->getBody();
        }

        $responseObject = $response->decodeJson();
        $key = $extra['indirection']['parse']['key'] ?? false;

        if ($key === false) {
            // format=json but no key specified
            return json_encode($responseObject);
        }

        // Look for a succession of possibly nested keys within a recursive array from the JSON object
        do {
            $index = is_array($key) ? key($key) : $key;
            if (! isset($responseObject[$index])) {
                break;
            }

            // Go one level deeper
            $responseObject = $responseObject[$index];

            if (! is_array($key)) {
                break;
            }

            $key = $key[$index] ?? false;
        } while (is_array($key) || is_string($key));

        return is_array($responseObject) ? json_encode($responseObject) : $responseObject;
    }
```

Reply:
What change is that?

Reply:
It does _bail out early_ thus reducing indentations and making it easier to read.

Reply:
I leave it 100% up to you.